java/ql/src/semmle/code/java/Expr.qll

```diff
@@ -653,11 +653,23 @@ class LongLiteral extends Literal, @longliteral {
 
 /** A floating point literal. For example, `4.2f`. */
 class FloatingPointLiteral extends Literal, @floatingpointliteral {
+  /**
+   * Gets the value of this literal as CodeQL 64-bit `float`. The value will
+   * be parsed as Java 32-bit `float` and then converted to a CodeQL `float`.
+   */
+  float getFloatValue() { result = getValue().toFloat() }
+
   override string getAPrimaryQlClass() { result = "FloatingPointLiteral" }
 }
 
 /** A double literal. For example, `4.2`. */
 class DoubleLiteral extends Literal, @doubleliteral {
+  /**
+   * Gets the value of this literal as CodeQL 64-bit `float`. The result will
+   * have the same effective value as the Java `double` literal.
```

Comment:
Added this to clarify how values which are not precise Java `double`s respectively CodeQL `float`s will be parsed.
That is, for example `1.55555555555555555555d` which is effectively `1.5555555555555556d` will also have that as result of this predicate.